### PR TITLE
fix(KFLUXBUGS-1498): expose the fbc opt in flag

### PR DESCRIPTION
**Pull request by happybhati**
* Added a new result `isFbcOptIn`, to expose the FBC opt-in status.

* Extracted and stored fbc_opt_in from the `genericResult` to make the opt-in status available for downstream tasks.

Note: This change will expose the `fbc_opt_in` flag only in case of success **until** Johnny's change for [KFLUXBUGS-1278](https://issues.redhat.com/browse/KFLUXBUGS-1278) gets merged. Once that is done we can see the flag even in case of IR failure.

### CODE REVIEW

File: tasks/add-fbc-contribution/README.md
Comment (mmalina) [inline tasks/add-fbc-contribution/README.md]: This should be 3.3.0.

File: tasks/add-fbc-contribution/add-fbc-contribution.yaml
Comment (mmalina) [inline tasks/add-fbc-contribution/add-fbc-contribution.yaml]: I wonder if we could also print some useful message here. I think the motivation for adding this was so that maybe users have a better understanding why their fbc update was not published. This is controlled by the `mustPublishIndexImage` - `isFbcOptIn` is just one factor.

So what if we added a check like: 
```sh
if $mustPublishIndexImage = true; then
  echo Index image will be published
elif $fbc_opt_in = false; then
  echo Index image will not be published because fbc opt in is set to false <where is this? in pyxis?>
else
  echo Index image will not be published, because <what are the other reasons that it's not published? because it's a staging release?>
fi
```

@theflockers WDYT?

Comment (theflockers) [reply tasks/add-fbc-contribution/add-fbc-contribution.yaml]: I think it makes sense. As you said, users have no clue why the publishing did not happen and maybe only exposing the value wouldn't help much.

I like the suggestion.

Comment (happybhati) [reply tasks/add-fbc-contribution/add-fbc-contribution.yaml]: Yes, I completely agree with the suggestion to clarify why the index image is not published for the user.

Statement from Slack thread: "When you want to have your index in production, you need the **stagedIndex: false** (or not set) and the fbcOptIn flag flipped in **pyxis**."


Do you think it looks any better? @mmalina @theflockers 
```
    if [ "$(jq -r '.genericResult | fromjson' <<< "${results}" | jq -r '.publish_index_image')" = "true" ]; then
      echo "Index image will be published."
    elif [ "$(jq -r '.genericResult | fromjson' <<< "${results}" | jq -r '.fbc_opt_in')" = "false" ]; then
      echo "Index image will not be published because FBC opt-in is set to false in Pyxis."
    elif [ "${staged_index}" = "true" ]; then
      echo "Index image will not be published because this is a staging release."
    else
      echo "Index image will not be published for an unspecified reason."
    fi
```

File: tasks/add-fbc-contribution/README.md
Comment (happybhati) [reply tasks/add-fbc-contribution/README.md]: Fixed, thanks :)

File: tasks/add-fbc-contribution/add-fbc-contribution.yaml
Comment (happybhati) [reply tasks/add-fbc-contribution/add-fbc-contribution.yaml]: I added some statements as suggested; please let me know if any corrections are needed. Thanks!

Comment (mmalina) [reply tasks/add-fbc-contribution/add-fbc-contribution.yaml]: I think the current version looks good. But I'd like @theflockers 's input as well.

**Review by mmalina — DISMISSED** [review]

File: tasks/add-fbc-contribution/add-fbc-contribution.yaml
Comment (mmalina) [inline tasks/add-fbc-contribution/add-fbc-contribution.yaml]: This should be 3.3.0

Comment (happybhati) [reply tasks/add-fbc-contribution/add-fbc-contribution.yaml]: Fixed, thanks!

**Review by mmalina — DISMISSED** [review]

File: tasks/add-fbc-contribution/add-fbc-contribution.yaml
Comment (theflockers) [reply tasks/add-fbc-contribution/add-fbc-contribution.yaml]: Also looks good to me.

**Review by theflockers — DISMISSED** [review]

**Review by johnbieren — COMMENTED** [review]
Code LGTM, I think we shouldn't merge until my 'never fail' PR is merged. It also wouldn't hurt to add a test case for the new result

**Review by johnbieren — DISMISSED** [review]

**Comment by johnbieren** [comment]
Approved but I still think a test case for this would be nice

**Comment by happybhati** [comment]
> Approved but I still think a test case for this would be nice

Thanks for approving. 
@johnbieren Are you suggesting a separate test or a check in the `test-add-fbc-contribution` test?

**Comment by johnbieren** [comment]
> > Approved but I still think a test case for this would be nice
> 
> Thanks for approving. @johnbieren Are you suggesting a separate test or a check in the `test-add-fbc-contribution` test?

Whichever feels better to you, as long as it is tested somehow

File: tasks/add-fbc-contribution/tests/test-add-fbc-contribution.yaml
Comment (theflockers) [inline tasks/add-fbc-contribution/tests/test-add-fbc-contribution.yaml]: I guess here should be `$(params.isFbcOptIn)` and `$(params.mustPublishIndexImage)`

Comment (happybhati) [reply tasks/add-fbc-contribution/tests/test-add-fbc-contribution.yaml]: Yes, thanks; that was a good catch. I have also updated the mock results, and now everything looks good.

File: tasks/add-fbc-contribution/tests/test-add-fbc-contribution.yaml
Comment (johnbieren) [inline tasks/add-fbc-contribution/tests/test-add-fbc-contribution.yaml]: This is a static test. How do we not know if it will be true or false? Same for fbc_opt_in

**Review by johnbieren — APPROVED** [review]

File: tasks/add-fbc-contribution/tests/test-add-fbc-contribution.yaml
Comment (happybhati) [reply tasks/add-fbc-contribution/tests/test-add-fbc-contribution.yaml]: Yes, I agree, those were unnecessary checks; thanks for your suggestion and approval.

**Pull request merged**
